src/lib/mongoBackend/mongoRegistrationCreate.cpp

```diff
@@ -114,16 +114,26 @@ static void setContextRegistrationVector(ngsiv2::Registration* regP, mongo::BSON
 
   for (unsigned int eIx = 0; eIx < regP->dataProvided.entities.size(); ++eIx)
   {
-    ngsiv2::EntID* eP = &regP->dataProvided.entities[eIx];
+    ngsiv2::EntID* eP                  = &regP->dataProvided.entities[eIx];
+    std::string    entityId            = (eP->id          != "")? eP->id : eP->idPattern;
+    std::string    entityIsPattern     = (eP->idPattern   != "")? "true" : "false";
+    std::string    entityType          = (eP->type        != "")? eP->type : eP->typePattern;
+    bool           entityTypeIsPattern = (eP->typePattern != "")? true : false;
```

Comment:
typePattern is not yet supported. Thus, this is not needed (I'd suggest to commmented out for the future, with a FIXME mark)

Reply:
In addition the block:

```
if (eP->type == "")
{
   entities.append(BSON(REG_ENTITY_ID << eP->id));
}
```

should be kept.

Reply:
It is kind of supported ...
If type is omitted, it equals typePattern == `.*`.
I use the variable `entityTypeIsPattern` for the logic in the block right after this line.

Reply:
I'd prefer to minimize the changes in this part to only the needed for idPattern `.*`.

Reply:
Fixed in 5d5e0b575